### PR TITLE
[WIP]allow harvesters transfer to other creeps

### PR DESCRIPTION
**Pull request by Nevrdid**

### CODE REVIEW

File: src/role_harvester.js [hunk 1]
```diff
@@ -102,6 +102,10 @@ roles.harvester.preMove = function(creep, directions) {
   if (directions && creep.memory.routing.reverse) {
     directions.direction = directions.backwardDirection;
   }
+  if (directions.backwardDirection && reverse && roles.scout.isStuck(creep)) {
```
Comment (TooAngel) [inline src/role_harvester.js]: The `isStuck` method should be on the creep, not on the scout.

Comment (Nevrdid) [reply src/role_harvester.js]: 👍

Comment (Nevrdid) [reply src/role_harvester.js]: 👍 yeah, was firstly made for period when harvesters are multiples in rooms and stuck themself often as there are harvester full of energy which wanna go to construction site near the controller and also other carrys trying to gonna fill extensions near container whitch are empty. not even sure it's the better solution now x) i'll WIP it as i'll not can work alot on next days.

**Comment by Nevrdid** [comment]
👍 yeah, was firstly made for period when harvesters are multiples in rooms and stuck themself often as there are harvester full of energy which wanna go to construction site near the controller and also other carrys trying to gonna fill extensions near container whitch are empty. not even sure it's the better solution now x) i'll WIP it as i'll not can work alot on next days.

**Pull request closed without merging**